### PR TITLE
Made the database for project class. Made activity class. Data is bei…

### PR DESCRIPTION
**Pull request by TalhaAsjad**
…ng inserted at head and tail, deleted from head and tail and being displayed.

### CODE REVIEW

**Review by haseebuchiha — APPROVED** [review]

File: CustomeDB/Activity.cpp [hunk 1]
```diff
@@ -11,12 +11,34 @@ void Activity::setData(string name, int id, double salary, string project, strin
 	emp.setAddress(address);
 }
 
+void Activity::setData(string name, int id, string description, int client_id, int money_earned, string deadline, string complete_date) {
```
Comment (ParanoidBat) [inline CustomeDB/Activity.cpp]: setData is an ambiguous function name. specify in name, what class are you setting data for. like: setDataProject. Likewise there are functions in other places that have same issue

Comment (ParanoidBat) [reply CustomeDB/Activity.cpp]: plus. change money_earned type to string. in its respective class as well and any other places such setter/getter

File: CustomeDB/LinkedList.cpp [hunk 2]
```diff
@@ -120,6 +131,21 @@ void LinkedList::addAtEnd(string name, int id, string phone, string address, int
...
+    if (prjhead == NULL) {
+        prjhead = temp;
+    }
+    else {
+        prjtail = prjhead;
+        while (prjtail->next != NULL) {
+            prjtail = prjtail->next;
+        }
+        prjtail->next = temp;
+        prjtail = temp;
```
Comment (ParanoidBat) [inline CustomeDB/LinkedList.cpp]: why are both the iterator(prjtail) and its next pointing to same object? and why isn't temp's next NULL ?

File: CustomeDB/CustomeDB.cpp [hunk 3]
```diff
@@ -2,13 +2,31 @@
...
 {
 
+    Activity test;
     
-    //Test ();
+    for (int i = 1; i < 6; i++) {
+        test.setData("Talha", i, "fuckboi", 69, "9000000", "3 hours", "1/25/2021");
+        test.enterToDatabasePrj();
+    }
+    //test.deleteHeadProjectDatabase();
```
Comment (haseebuchiha) [inline CustomeDB/CustomeDB.cpp]: Why is there code in main file???

File: CustomeDB/LinkedList.cpp [hunk 4]
```diff
@@ -216,23 +314,93 @@ void LinkedList::deleteById(int id) {
...
+                cout << " enter the new deadline : ";
+                cin >> deadline;
+                temp->prj.setdeadline(deadline);
+            }
+            else {
+                cout << "Choose betweem given numbers \n";
+            }
+        }
+        if (!found) {
+            cout << "This id is not present in database. \n";
```
Comment (haseebuchiha) [inline CustomeDB/LinkedList.cpp]: Why did you add code for input directly inside linkedlist class??? If you want to handle input you can handle it inside Activity class only

File: CustomeDB/LinkedList.cpp [hunk 5]
```diff
@@ -271,137 +454,114 @@ void LinkedList::updateSalary(int id, string name, double new_salary) {
...
+//            while (index != NULL) {
+//                if (cur->getClient().getname() < index->getClient().getname()) {
+//                    swapNodes(cur, index);
+//                }
+//                index = index->next;
+//            }
+//            cur = cur->next;
+//        }
+//    }
+//}
```
Comment (haseebuchiha) [inline CustomeDB/LinkedList.cpp]: Remove the comments

File: CustomeDB/LinkedList.cpp [hunk 6]
```diff
@@ -450,25 +620,25 @@ void LinkedList::reverseLinkedList() {
...
+//    else {
+//        while (cur != NULL) {
+//            temp = cur->next;
+//            cur->next = prev;
+//            prev = cur;
+//            cur = temp;
+//        }
+//        clienthead = prev;
+//    }
+//}
```
Comment (haseebuchiha) [inline CustomeDB/LinkedList.cpp]: Remove the comments

File: CustomeDB/LinkedList.cpp [hunk 7]
```diff
@@ -415,13 +575,23 @@ size_t LinkedList::listSize() {
...
+//
+//    Node* cur = clienthead;
+//
+//    size_t counter = 0;
+//    while (cur != NULL) {
+//        cur = cur->next;
+//        counter++;
+//    }
+//    return counter;
+//}
```
Comment (haseebuchiha) [inline CustomeDB/LinkedList.cpp]: Remove the comments

File: CustomeDB/LinkedList.h [hunk 8]
```diff
@@ -9,50 +9,63 @@ class LinkedList {
...
-    void sortByNameAes();
+    //void sortByNameAes();
 
-    void sortByNameDec();
-    void sortByNameDec();
 
+    //void sortByNameDec();
+ 
+
+    //size_t listSize();
```
Comment (haseebuchiha) [inline CustomeDB/LinkedList.h]: Remove comments

File: CustomeDB/LinkedList.cpp [hunk 9]
```diff
@@ -216,23 +314,93 @@ void LinkedList::deleteById(int id) {
...
+//        cout << cur->getEmployee().getSalary() << " ";
+//        cout << cur->getEmployee().getProject() << " ";
+//        cout << cur->getEmployee().getJoiningDate() << " ";
+//        cout << cur->getEmployee().getPhone() << " ";
+//        cout << cur->getEmployee().getAddress() << " \n";
+//        cur = cur->next;
+//    }
+//    cout << endl;
+//}
+
```
Comment (haseebuchiha) [inline CustomeDB/LinkedList.cpp]: Remove Comments

File: CustomeDB/LinkedList.cpp [hunk 10]
```diff
@@ -133,41 +163,109 @@ void LinkedList::deleteAtHead() {
...
+//                temp = temp->next;
+//            }
+//            temp->next = cur->next;
+//            delete cur;
+//        }
+//        else {
+//            cout << "Id doesnt exist in linked list \n";
+//        }
+//    }
+//}
```
Comment (haseebuchiha) [inline CustomeDB/LinkedList.cpp]: Remove Comments

File: CustomeDB/LinkedList.cpp [hunk 11]
```diff
@@ -133,41 +163,109 @@ void LinkedList::deleteAtHead() {
...
+//        cout << "There is no linked list to delete" << endl;
+//    }
+//    else {
+//        Node* temp = clienthead;
+//        clienthead = clienthead->next;
+//
+//        delete temp;
+//    }
+//}
+
```
Comment (haseebuchiha) [inline CustomeDB/LinkedList.cpp]: Remove Comments

**Review by haseebuchiha — CHANGES_REQUESTED** [review]
Fix all the commented changes

**Pull request merged**